diff --git a/pexpert/arm/common/pe_serial.c b/pexpert/arm/common/pe_serial.c
--- a/pexpert/arm/common/pe_serial.c
+++ b/pexpert/arm/common/pe_serial.c
@@ -28,8 +28,9 @@
  */
 
 #include <mach/mach_types.h>
+#include <kern/debug.h>
 #include <pexpert/pexpert.h>
 #include <pexpert/arm/boot.h>
 
 extern int pe_initialized;
 
diff --git a/osfmk/arm/machine_routines.h b/osfmk/arm/machine_routines.h
--- a/osfmk/arm/machine_routines.h
+++ b/osfmk/arm/machine_routines.h
@@ -44,6 +44,9 @@
 __BEGIN_DECLS
 // why
 void bcopy_phys(addr64_t src64, addr64_t dst64, vm_size_t bytes);
+
+/* Defined in osfmk/arm/machine_routines_asm.s */
+void Halt_system(void);
 
 #ifdef XNU_KERNEL_PRIVATE
 
