diff --git a/bsd/kern/kdebug.c b/bsd/kern/kdebug.c
index dfc3873..75cc604 100644
--- a/bsd/kern/kdebug.c
+++ b/bsd/kern/kdebug.c
@@ -75,6 +75,8 @@
 /* XXX should have prototypes, but Mach does not provide one */
 void task_act_iterate_wth_args(task_t, void(*)(thread_t, void *), void *);
 int cpu_number(void);	/* XXX <machine/...> include path broken */
+void disable_preemption(void);
+void enable_preemption(void);
 
 /* XXX should probably be static, but it's debugging code... */
 int kdbg_read(user_addr_t, size_t *, vnode_t, vfs_context_t);
diff --git a/pexpert/arm/common/pe_init.c b/pexpert/arm/common/pe_init.c
index 56f03c5..36c8b2e 100644
--- a/pexpert/arm/common/pe_init.c
+++ b/pexpert/arm/common/pe_init.c
@@ -31,6 +31,7 @@
  */
 
 #include <sys/types.h>
+#include <sys/systm.h>
 #include <mach/vm_param.h>
 #include <machine/machine_routines.h>
 #include <pexpert/protos.h>
diff --git a/pexpert/pexpert/arm/protos.h b/pexpert/pexpert/arm/protos.h
index 61bd7bb..62da0af 100755
--- a/pexpert/pexpert/arm/protos.h
+++ b/pexpert/pexpert/arm/protos.h
@@ -101,6 +101,8 @@ static inline uint8_t regread8(uint8_t addr)
 void cnputc(char);
 int switch_to_serial_console(void);
 void switch_to_old_console(int);
+boolean_t console_is_serial(void);
+void panic_ui_initialize(const unsigned char *system_clut);
 
 
 /*
