diff --git a/bsd/kern/kern_exec.c b/bsd/kern/kern_exec.c
index eae4a73..b989b6e 100644
--- a/bsd/kern/kern_exec.c
+++ b/bsd/kern/kern_exec.c
@@ -167,6 +167,7 @@ void vfork_exit(proc_t p, int rv);
 int setsigvec(proc_t, thread_t, int, struct __kern_sigaction *, boolean_t in_sigstart);
 extern void proc_apply_task_networkbg_internal(proc_t, thread_t);
 int task_set_cpuusage(task_t task, uint64_t percentage, uint64_t interval, uint64_t deadline, int scope);
+extern void pal_dbg_set_task_name(task_t task);
 
 /*
  * Mach things for which prototypes are unavailable from Mach headers
diff --git a/osfmk/vm/vm_pageout.c b/osfmk/vm/vm_pageout.c
index 2bfd6e7..8eec348 100644
--- a/osfmk/vm/vm_pageout.c
+++ b/osfmk/vm/vm_pageout.c
@@ -3022,7 +3022,7 @@ uint32_t vm_pageout_considered_page_last = 0;
  * called once per-second via "compute_averages"
  */
 void
-compute_pageout_gc_throttle()
+compute_pageout_gc_throttle(void *arg __unused)
 {
 	if (vm_pageout_considered_page != vm_pageout_considered_page_last) {
 
diff --git a/pexpert/pexpert/protos.h b/pexpert/pexpert/protos.h
index d544644..f69a2b8 100644
--- a/pexpert/pexpert/protos.h
+++ b/pexpert/pexpert/protos.h
@@ -42,7 +42,7 @@
 
 //------------------------------------------------------------------------
 // from ppc/misc_protos.h
-extern void printf(const char *fmt, ...);
+extern int printf(const char *fmt, ...);
 
 extern void interrupt_enable(void);
 extern void interrupt_disable(void);
