diff --git a/libkern/mkext.c b/libkern/mkext.c
index 86238fc..1b47b30 100644
--- a/libkern/mkext.c
+++ b/libkern/mkext.c
@@ -47,6 +47,7 @@ mkext_adler32(uint8_t *buf, int32_t len)
     int k;
 
 #if defined _ARM_ARCH_6
+    extern unsigned long adler32_vec(unsigned long adler, unsigned long sum2, const uint8_t *buf, unsigned int len);
 
 	/* align buf to 16-byte boundary */
     while ((((uintptr_t)buf)&15)&&(len>0)) { /* not on a 16-byte boundary */
@@ -284,4 +285,3 @@ static void delete_node(struct encode_state *sp, int p)
 }
 
 #endif /* !KERNEL */
-
diff --git a/osfmk/kern/kmod.c b/osfmk/kern/kmod.c
index 160c4bd..69d81d2 100644
--- a/osfmk/kern/kmod.c
+++ b/osfmk/kern/kmod.c
@@ -85,7 +85,7 @@ extern void proc_selfname(char * buf, int size);
             __FUNCTION__, procname);  \
     } while (0)
 
-#if __i386__
+#if __i386__ || __arm__
 // in libkern/OSKextLib.cpp
 extern kern_return_t kext_get_kmod_info(
     kmod_info_array_t      * kmod_list,
@@ -93,7 +93,7 @@ extern kern_return_t kext_get_kmod_info(
 #define KMOD_MIG_UNUSED
 #else
 #define KMOD_MIG_UNUSED __unused
-#endif /* __i386__ */
+#endif /* __i386__ || __arm__ */
 
 
 /*********************************************************************
diff --git a/pexpert/arm/pe_imx53.c b/pexpert/arm/pe_imx53.c
index 5153a3a..f450a6a 100644
--- a/pexpert/arm/pe_imx53.c
+++ b/pexpert/arm/pe_imx53.c
@@ -280,8 +280,8 @@ void imx53_framebuffer_init(void)
     /*
      * This is an emulated framebuffer. 
      */
-    void *framebuffer = pmap_steal_memory(1024 * 768 * 4);
-    void *framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
+    vm_offset_t framebuffer = (vm_offset_t)pmap_steal_memory(1024 * 768 * 4);
+    vm_offset_t framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
 
     uint32_t depth = 4;
     uint32_t width = 1024;
@@ -297,7 +297,7 @@ void imx53_framebuffer_init(void)
     PE_state.video.v_depth = 2 * (8);   // 16bpp
 
     kprintf(KPRINTF_PREFIX "fake framebuffer initialized\n");
-    bzero(framebuffer, (pitch * height));
+    bzero((void *)framebuffer, (pitch * height));
 
     initialize_screen((void *) &PE_state.video, kPETextMode);
 }
diff --git a/pexpert/arm/pe_omap335x.c b/pexpert/arm/pe_omap335x.c
index f65a572..8801492 100644
--- a/pexpert/arm/pe_omap335x.c
+++ b/pexpert/arm/pe_omap335x.c
@@ -349,8 +349,8 @@ void Omap3_framebuffer_init(void)
     /*
      * This is an emulated framebuffer. 
      */
-    void *framebuffer = pmap_steal_memory(1024 * 768 * 4);
-    void *framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
+    vm_offset_t framebuffer = (vm_offset_t)pmap_steal_memory(1024 * 768 * 4);
+    vm_offset_t framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
 
     uint32_t depth = 4;
     uint32_t width = 1024;
@@ -366,7 +366,7 @@ void Omap3_framebuffer_init(void)
     PE_state.video.v_depth = 2 * (8);   // 16bpp
 
     kprintf(KPRINTF_PREFIX "fake framebuffer initialized\n");
-    bzero(framebuffer, (pitch * height));
+    bzero((void *)framebuffer, (pitch * height));
 
     char tempbuf[16];
     initialize_screen((void *) &PE_state.video, kPETextMode);
diff --git a/pexpert/arm/pe_omap3430.c b/pexpert/arm/pe_omap3430.c
index b027578..f61b17c 100644
--- a/pexpert/arm/pe_omap3430.c
+++ b/pexpert/arm/pe_omap3430.c
@@ -474,10 +474,10 @@ void Omap3_framebuffer_init(void)
     /*
      * Allocate framebuffer 
      */
-    void *framebuffer = pmap_steal_memory(lcd_width * lcd_width * 4);
-    void *framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
-    bzero(framebuffer, lcd_width * lcd_height * 4);
-    kprintf(KPRINTF_PREFIX "software framebuffer at %p\n", framebuffer);
+    vm_offset_t framebuffer = (vm_offset_t)pmap_steal_memory(lcd_width * lcd_width * 4);
+    vm_offset_t framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
+    bzero((void *)framebuffer, lcd_width * lcd_height * 4);
+    kprintf(KPRINTF_PREFIX "software framebuffer at %p\n", (void *)framebuffer);
 
 #if 0
     /*
diff --git a/pexpert/arm/pe_omap3530.c b/pexpert/arm/pe_omap3530.c
index 1cc6fbd..44ff6e1 100644
--- a/pexpert/arm/pe_omap3530.c
+++ b/pexpert/arm/pe_omap3530.c
@@ -502,10 +502,10 @@ void Omap3_framebuffer_init(void)
     /*
      * Allocate framebuffer 
      */
-    void *framebuffer = pmap_steal_memory(lcd_width * lcd_width * 4);
-    void *framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
-    bzero(framebuffer, lcd_width * lcd_height * 4);
-    kprintf(KPRINTF_PREFIX "software framebuffer at %p\n", framebuffer);
+    vm_offset_t framebuffer = (vm_offset_t)pmap_steal_memory(lcd_width * lcd_width * 4);
+    vm_offset_t framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
+    bzero((void *)framebuffer, lcd_width * lcd_height * 4);
+    kprintf(KPRINTF_PREFIX "software framebuffer at %p\n", (void *)framebuffer);
 
     /*
      * Set attributes in display controller 
@@ -546,7 +546,7 @@ void Omap3_framebuffer_init(void)
         /*
          * BootX like framebuffer. 
          */
-        memset(framebuffer, 0xb9, lcd_width * lcd_height * 4);
+        memset((void *)framebuffer, 0xb9, lcd_width * lcd_height * 4);
         initialize_screen((void *) &PE_state.video, kPEGraphicsMode);
     } else {
         initialize_screen((void *) &PE_state.video, kPETextMode);
diff --git a/pexpert/arm/pe_realview.c b/pexpert/arm/pe_realview.c
index e822eb3..d0e7627 100644
--- a/pexpert/arm/pe_realview.c
+++ b/pexpert/arm/pe_realview.c
@@ -332,8 +332,8 @@ void RealView_framebuffer_init(void)
      * The hardware demands a framebuffer, but the framebuffer has to be given
      * in a hardware address.
      */
-    void *framebuffer = pmap_steal_memory(1024 * 768 * 4);
-    void *framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
+    vm_offset_t framebuffer = (vm_offset_t)pmap_steal_memory(1024 * 768 * 4);
+    vm_offset_t framebuffer_phys = pmap_extract(kernel_pmap, framebuffer);
 
     uint32_t depth = 2;
     uint32_t width = 1024;
@@ -368,7 +368,7 @@ void RealView_framebuffer_init(void)
     PE_state.video.v_depth = 4 * (8);   // 16bpp
 
     kprintf(KPRINTF_PREFIX "framebuffer initialized\n");
-    bzero(framebuffer, (pitch * height));
+    bzero((void *)framebuffer, (pitch * height));
 
     char tempbuf[16];
     
@@ -376,7 +376,7 @@ void RealView_framebuffer_init(void)
         /*
          * BootX like framebuffer. 
          */
-        memset(framebuffer, 0xb9, PE_state.video.v_rowBytes * PE_state.video.v_height);
+        memset((void *)framebuffer, 0xb9, PE_state.video.v_rowBytes * PE_state.video.v_height);
         initialize_screen((void *) &PE_state.video, kPEGraphicsMode);
     } else {
 		initialize_screen((void *) &PE_state.video, kPETextMode);
